### PR TITLE
DSD-1303: Fix Slider onChangeEnd with value prop issue

### PR DESCRIPTION
**Pull request by jackiequach**
Fixes JIRA ticket [DSD-1303](https://jira.nypl.org/browse/DSD-1303)

## This PR does the following:

- Replaces `useEffect` with the `useStateWithDependencies` hook to update the `currentValue` state.

## How has this been tested?

Locally on Storybook

## Accessibility concerns or updates

-

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Two things:
1. Update the version in its Storybook doc page.
2. Add a new test or modify the "gets the current value through the onChangeEnd callback function" test so that the returned value from `onChangeEnd` get passed back to the `Slider` component in the `value` prop.

File: src/components/Slider/Slider.stories.mdx [hunk 1]
```diff
@@ -75,10 +75,10 @@ import DSProvider from "../../theme/provider";
...
 # Slider
 
-| Component Version | DS Version |
-| ----------------- | ---------- |
-| Added             | `0.25.4`   |
-| Latest            | `1.4.1`    |
+| Component Version | DS Version   |
+| ----------------- | ------------ |
+| Added             | `0.25.4`     |
+| Latest            | `Prerelease` |
```
Comment (EdwinGuzman) [inline src/components/Slider/Slider.stories.mdx:81]: I think the next version will be 1.4.2 but Marty should verify.

**Review by EdwinGuzman — APPROVED** [review]
Looks good! Thanks.

File: src/components/Slider/Slider.stories.mdx (hunk 1 above)
Comment (bigfishdesign13) [reply src/components/Slider/Slider.stories.mdx:81]: Yes, we can release this next week as `1.4.2`.

Comment (bigfishdesign13) [reply src/components/Slider/Slider.stories.mdx:81]: @jackiequach Actually, don't change this before merging.  I like the idea of using a generic value when udpates are made.  We can update to the proper version number as part of the release prep process.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Pull request merged**